homeassistant/helpers/entity.py

```diff
@@ -1009,7 +1009,7 @@ def _call_on_remove_callbacks(self) -> None:
     def add_to_platform_abort(self) -> None:
         """Abort adding an entity to a platform."""
 
-        self._platform_state = EntityPlatformState.NOT_ADDED
+        self._platform_state = EntityPlatformState.REMOVED
```

Comment:
This is called if `EntityPlatform` fails to add the entity. `EntityPlatform` does not retry, so we set it to `REMOVED` here.